src/functions.php

```diff
@@ -140,6 +140,32 @@ function isfile(string $path): Promise
     return filesystem()->isfile($path);
 }
 
+/**
+ * Does the specified path exist and is it a symlink?
+ *
+ * If the path does not exist the returned Promise will resolve
+ * to FALSE and will not reject with an error.
+ *
+ * @param string $path An absolute file system path
+ * @return Promise<bool>
+ */
+function islink(string $path): Promise
+{
+    return filesystem()->islink($path);
+}
+
+/**
+ * Retrieve the path's file permissions.
+ *
+ * @param string $path An absolute file system path
+ * @fails \Amp\Files\FilesystemException If the path does not exist
+ * @return Promise<string>
+ */
+function fileperms(string $path): Promise
```

Comment:
I'm not sure whether we should keep the PHP names or should use better names for each function.

Reply:
By keeping the same name you make it obvious that this function is an async equivalent. But if you want to name it differently, I'm fine with that.